pkg/api/v1/routes/handlers.go

```diff
@@ -79,8 +79,54 @@ func (r *Routes) serverConditionCreate(c *gin.Context) (int, *v1types.ServerResp
 		}
 	}
 
+	return r.conditionCreate(otelCtx, c, kind, serverID)
+}
+
+func (r *Routes) serverEnroll(c *gin.Context) (int, *v1types.ServerResponse) {
+	otelCtx, span := otel.Tracer(pkgName).Start(c.Request.Context(), "Routes.serverEnroll")
+	span.SetAttributes(
+		attribute.KeyValue{Key: "serverId", Value: attribute.StringValue(c.Param("id"))},
+		attribute.KeyValue{Key: "IP", Value: attribute.StringValue(c.Param("ip"))},
+		attribute.KeyValue{Key: "user", Value: attribute.StringValue(c.Param("user"))})
+	defer span.End()
+
+	serverID, err := uuid.Parse(c.Param("uuid"))
+	if err != nil {
+		r.logger.WithFields(logrus.Fields{
+			"serverID": c.Param("uuid"),
+		}).Info("bad serverID")
+
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: err.Error(),
+		}
+	}
+
+	// Creates a server record in FleetDB.
+	if err = r.repository.AddServer(c.Request.Context(), serverID, c.Param("ip"), c.Param("user"), c.Param("pwd")); err != nil {
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: err.Error(),
+		}
+	}
+
+	// Reuse the inventory for server enrollment.
+	// May change in the future.
+	kind := rctypes.Kind("inventory")
+	if !r.conditionKindValid(kind) {
```

Comment:
You don't need to test the validity of the kind here; we're not accepting untrusted external input. We're specifically using a valid condition `Kind`.

Reply:
> Do we have other tools currently (or planned) that would update the BMC info? If not it's probably better to keep this private.
@DoctorVin we do have systems that require to look up that data, and must know the namespace to look it up from.